GVFS/GVFS.Common/PlaceholderListDatabase.cs

```diff
@@ -167,6 +167,58 @@ public void GetAllEntries(out List<PlaceholderData> filePlaceholders, out List<P
             }
         }
 
+        public void GetAllEntries(
+            out Dictionary<string, PlaceholderListDatabase.PlaceholderData> placeholderDataByPath,
+            out List<PlaceholderData> folderPlaceholders)
+        {
+            try
+            {
+                Dictionary<string, PlaceholderListDatabase.PlaceholderData> filePlaceholdersFromDiskByPath = 

```

Comment:
What is the memory usage with this going to be if a user has the entire repo as placeholders?